### PR TITLE
Included extend() explanation in services documentation chapter

### PR DESCRIPTION
**Pull request by hvt**
Together with the Twig add global and filter example.

### CODE REVIEW

File: doc/services.rst [hunk 1]
```diff
@@ -161,6 +161,31 @@ using the ``protect`` method::
 
 Note that protected closures do not get access to the container.
 
+Modify services after definition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sometimes you want to alter a service after its definition. Pimple facilitates
+this by ``extend``-ing the already defined service.
```
Comment (stof) [inline doc/services.rst]: I would just say ``by extending``, without using a code block in the middle

File: doc/services.rst [hunk 2]
```diff
@@ -161,6 +161,31 @@ using the ``protect`` method::
...
+
+First argument of the ``extend`` method is the name of the service you want to
+modify. Second argument is a callable. This callable is executed with the service
+you want to alter as its first argument, the service container self is provided
```
Comment (stof) [inline doc/services.rst]: should be ``itself``, not ``self``

**Comment by GromNaN** [comment]
This is a feature of Pimple. You improve the Pimple documentation and link to it : http://pimple.sensiolabs.org/#modifying-services-after-definition

**Comment by stof** [comment]
Linking to the Pimple documentation is indeed a much better idea.

File: doc/services.rst (hunk 2 above)
Comment (hvt) [inline doc/services.rst]: Tried avoiding using `it` twice in consecutive sentences, but wil fix.

**Comment by hvt** [comment]
So why are there descriptions of `factory` and `protect`? These are both also Pimple features.

Besides, the `extend` example is already on the Twig provider page. Some explanation of what `extend` is and does, seems appropriate to me.

**Comment by fabpot** [comment]
Thank you @hvt.

**Pull request merged**